File: backend/src/main/java/com/equisoft/dca/backend/location/service/LocationService.java
package com.equisoft.dca.backend.location.service;

import java.util.List;
import java.util.Optional;

import javax.validation.Valid;

import com.equisoft.dca.backend.filesystem.model.FileData;
import com.equisoft.dca.backend.location.model.Location;

public interface LocationService {

	Optional<Location> findById(Integer id);

	List<Location> findAll();

	Location save(@Valid Location location);

	Location update(@Valid Location location);

	void deleteById(Integer id);

	List<FileData> getFiles(Integer id);
}

File: react/src/lib/services/LocationService.ts
import FileData from '../domain/entities/FileData';
import Location from '../domain/entities/Location';
import LocationRepository from '../domain/repositories/LocationRepository';

export default class LocationService {
    constructor(private repository: LocationRepository) {
    }

    async add(location: Location): Promise<Location> {
        return this.repository.add(location);
    }

    async update(location: Location): Promise<void> {
        return this.repository.update(location);
    }

    async delete(id: string): Promise<void> {
        return this.repository.delete(id);
    }

    async findAll(): Promise<Location[]> {
        return this.repository.findAll();
    }

    async findById(id: string): Promise<Location> {
        return this.repository.findById(id);
    }

    async getFiles(id: string): Promise<FileData[]> {
        return this.repository.getFiles(id);
    }
}